src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobContainerClientProxy.java

```diff
@@ -93,4 +105,56 @@ public void upload(
             throw ex;
         }
     }
+
+    public void moveBlob(
+        BlobClient sourceBlob,
+        String destinationContainer,
+        TargetStorageAccount targetStorageAccount
+    ) {
+        try {
+            var blobName = sourceBlob.getBlobName();
+            logger.info("Move from {}   to Container: {}", sourceBlob.getBlobUrl(),
+                destinationContainer);
+
+            String sasToken = sourceBlob
+                .generateSas(
+                    new BlobServiceSasSignatureValues(
+                        OffsetDateTime
+                            .of(LocalDateTime.now().plus(5, ChronoUnit.MINUTES), ZoneOffset.UTC),
+                        new BlobContainerSasPermission().setReadPermission(true)
+                    )
+                );
+            final BlockBlobClient targetBlob =
+                get(targetStorageAccount, destinationContainer)
+                    .getBlobClient(blobName)
+                    .getBlockBlobClient();
+
+            var start = System.nanoTime();
+            final SyncPoller<BlobCopyInfo, Void> poller =
+                targetBlob.beginCopy(sourceBlob.getBlobUrl() + "?" + sasToken, Duration.ofSeconds(2));
+            PollResponse<BlobCopyInfo> pollResponse = poller.waitForCompletion(Duration.ofMinutes(10));
```

Comment:
We should check status of PollResponse and continue depending on the result - SUCCESSFULLY_COMPLETED or FAILED.

Reply:
I though it may stay on pending status forever that is why I add timeout. Processor side is not doing anything if it the status is not SUCCESS.

Reply:
Maybe there should be some retry policy? Anyway failure must be handled.